### PR TITLE
BE-1-init-application

### PR DESCRIPTION
**Pull request by Randayy**

### CODE REVIEW

File: App/.env.sample [hunk 1]
```diff
@@ -0,0 +1,3 @@
+SECRET_KEY=
```
Comment (Ghibliliker) [inline App/.env.sample]: u use all of them?

File: App/main.py
Comment (Ghibliliker) [inline App/main.py:1]: rename to "app"

File: App/main.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from fastapi import FastAPI
+
+app = FastAPI()
+
+@app.get("/")
+async def health_check():
```
Comment (Ghibliliker) [inline App/main.py]: move it from main

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+fastapi
+redis
+types-redis
+uvicorn
+httpx
+pytest
```
Comment (Ghibliliker) [inline requirements.txt]: indicate versions

File: tests/test_main.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from fastapi.testclient import TestClient
+from App.main import app
+
+client = TestClient(app)
```
Comment (Ghibliliker) [inline tests/test_main.py]: add it like as fixture

File: App/.env.sample (hunk 1 above)
Comment (Randayy) [reply App/.env.sample]: not now , but in future I can

**Review by Makkrich — CHANGES_REQUESTED** [review]
All Pavlo's comments +
- Define whole project structure
- Split requirements into requirements.txt and requirements.dev.txt
- You dont use HOST and PORT env vars

File: App/main.py
Comment (Randayy) [reply App/main.py:1]: renamed

File: App/main.py (hunk 2 above)
Comment (Randayy) [reply App/main.py]: moved to routes/routes.py

File: tests/test_main.py (hunk 4 above)
Comment (Randayy) [reply tests/test_main.py]: added

File: requirements.txt (hunk 3 above)
Comment (Randayy) [reply requirements.txt]: included

**Comment by Randayy** [comment]
@Fan4ik20 > * Define whole project structure
it will be okay ? or how it should look
app
├── __init__.py
├── core
├── db
├── main.py
├── routes
│   ├── __init__.py
│   └── routes.py
├── schemas
├── services
├── settings.py
└── utils

**Comment by Randayy** [comment]
Fixed every Pavlo`s comment and that 

- Split requirements into requirements.txt and requirements.dev.txt

- You dont use HOST and PORT env vars

How should project structure looks like?

@Fan4ik20 @Ghibliliker @Artemooon @Fhockk

**Comment by Fhockk** [comment]
image
The documentation contains a photo of what the project structure should look like, are there any specific questions?

**Comment by Fhockk** [comment]
You have folder "App" it is an error. move the files from it to the root folder, and also create the missing folders and __init__ files inside

File: settings.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from dotenv import load_dotenv
+import os
+
+load_dotenv()
+
+HOST = os.getenv("HOST")
+PORT = os.getenv("PORT")
```
Comment (Fhockk) [inline settings.py]: please move it to class (check pydantic_settings)

File: settings.py
Comment (Fhockk) [inline settings.py:1]: Wrong directory and rename it, move it to app/core/config.py

File: __init__.py
Comment (Fhockk) [inline __init__.py:1]: delete it, dont need in root directory

File: requirements.dev.txt
Comment (Fhockk) [inline requirements.dev.txt:1]: if u split requirements to separate files, you need do in this way:
requirements.txt:
fastapi==x.x.x
pydantic_settings==x.x.x 
uvicorn==x.x.x
pydantic==x.x.x

requirements_dev.txt:
-r requirements.txt
pytest=x.x.x
httpx==x.x.x
Other stuff

**Review by Fhockk — CHANGES_REQUESTED** [review]

**Comment by Fhockk** [comment]
When i said delete "App" i didnt mean "app". You had both, but first was wrong

**Comment by Randayy** [comment]
added folder app,
added /core/config.py,
deleted __init__.py in root directory,
changed requirements.txt and requirements.dev.txt

File: app/tests/test_main.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import pytest
+from fastapi.testclient import TestClient
+from utils.main import app
+
+@pytest.fixture
```
Comment (Fhockk) [inline app/tests/test_main.py]: move fixture to conftest

File: app/tests/test_main.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    response = client.get("/")
+    assert response.status_code == 200
+    assert response.json() == { 
+            "status_code": 200,
+            "detail": "ok",
+            "result": "working"
+            }
+    
+if __name__ == "__main__":
+    pytest.main()
```
Comment (Fhockk) [inline app/tests/test_main.py]: why do u need that?

File: app/tests/__init__.py
Comment (Fhockk) [inline app/tests/__init__.py:1]: Move test directory from "app". They have one level

File: app/utils/main.py
Comment (Fhockk) [inline app/utils/main.py:1]: move main outside utills func. its in app directory. And where do u start ur app with ur envs?

**Review by Fhockk — CHANGES_REQUESTED** [review]

File: app/utils/main.py
Comment (Randayy) [reply app/utils/main.py:1]: moved

File: app/tests/__init__.py
Comment (Randayy) [reply app/tests/__init__.py:1]: moved

File: app/tests/test_main.py (hunk 6 above)
Comment (Randayy) [reply app/tests/test_main.py]: moved

File: app/routers/routes.py
Comment (Fhockk) [inline app/routers/routes.py:1]: its good to name in some logic way: health_check_route (for example)

File: tests/test_main.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
+from .conftest import client
```
Comment (Fhockk) [inline tests/test_main.py]: delete it, u dont need to import fixture if it initialized in conftest

File: app/main.py [hunk 9]
```diff
@@ -0,0 +1,8 @@
+from fastapi import FastAPI
+from .routers.routes import router
+
+
+app = FastAPI()
+
+app.include_router(router)
+
```
Comment (Fhockk) [inline app/main.py:9]: where is ur logic to start ur app by envs credentials?
and it is good practise to add DEBUG value to env and use it with "reload" value

**Review by Fhockk — CHANGES_REQUESTED** [review]
> Describe how your application starts in README.md

File: tests/test_main.py (hunk 8 above)
Comment (Randayy) [reply tests/test_main.py]: deleted

File: app/routers/routes.py
Comment (Randayy) [reply app/routers/routes.py:1]: renamed

Comment (Fhockk) [reply app/routers/routes.py:1]: push commit then

File: app/main.py (hunk 9 above)
Comment (Randayy) [reply app/main.py:9]: added DEBUG to env and used it with reload

**Comment by Fhockk** [comment]
> Describe how your application starts in README.md

still dont have it

**Comment by Randayy** [comment]
made to run app use this:
```bash
python -m app.main

**Pull request merged**